packages/vite/src/node/plugins/resolve.ts

```diff
@@ -530,9 +534,11 @@ function tryResolveFile(
   targetWeb: boolean,
   tryPrefix?: string,
   skipPackageJson?: boolean,
+  skipTsExtension?: boolean,
 ): string | undefined {
-  if (isFileReadable(file)) {
-    if (!fs.statSync(file).isDirectory()) {
+  const stat = fs.statSync(file, { throwIfNoEntry: false })
```

Comment:
I don't know a lot with vite but doesn't this need a try catch as it would throw an error for virtual modules
See [this error an Astro user encountered](https://discord.com/channels/830184174198718474/1070680332283674675/1070681049983623238)